src/Microsoft.Health.Dicom.Core/Features/Store/StoreService.cs

```diff
@@ -184,19 +184,10 @@ TelemetryClient telemetryClient
 
             if (_enableDropInvalidDicomJsonMetadata)
             {
-                var identifier = dicomDataset.ToInstanceIdentifier();
+                DropInvalidMetadata(storeValidatorResult, dicomDataset);
 
-                foreach (DicomTag tag in storeValidatorResult.InvalidTagErrors.Keys)
-                {
-                    // drop invalid metadata
-                    dicomDataset.Remove(tag);
-
-                    string message = storeValidatorResult.InvalidTagErrors[tag].Error;
-                    _telemetryClient.ForwardLogTrace(
-                        $"{message}. This attribute will not be present when retrieving study, series, or instance metadata resources, nor can it be used in searches." +
-                        "However, it will still be present when retrieving study, series, or instance resources.",
-                        identifier);
-                }
+                // set warning code if none set yet
+                warningReasonCode ??= WarningReasonCodes.DatasetHasValidationWarnings;
```

Comment:
In case of _enableDropInvalidDicomJsonMetadata == true, we are not using this warning code in the response builder. It is only set when the flag is off. Also only when the warning code is IndexedDicomTagHasMultipleValues we set the warning header. Do we want to set the header if there is error in the attribute too?